### DISCUSSION TITLE
Ignore Regex Function while linting

### DISCUSSION THREAD

**Discussion by kferrone** [terraform-linters/tflint#2127] [Q&A] [answered]
I am totally stuck with tflint. It really wants to check that a regex function is working but the value the regex function is checking is coming from a file. It seems tflint isn't checking in that file and therefore has no idea what that value is. Now I keep getting this error: 
```
$ tflint -c $TFLINT_FILE
Failed to check ruleset; failed to check "aws_api_gateway_domain_name_invalid_security_policy" rule: main.tf:34,16-22: Error in function call; Call to function "regex" failed: pattern did not match any part of the given string., and 2 other diagnostic(s)
```

I discovered this rule is just noise: `aws_api_gateway_domain_name_invalid_security_policy`. It gets stuck on the regex thing and can't process rules after that. Everything works perfect if I remove the regex function. The terraform module itself works totally fine as well, ie validate, plan, and apply all work fine. 

I can't google the error either. I just have not come up with anyone else having this issue. 

How can I get tflint to simply ignore the regex function? Can I just ignore the variable somehow and tell TFLINT to not try any rule at all on this variable?  

The block in question looks like this: 
```hcl
locals {
  integrations = flatten([
    for path, methods in local.body.paths : [
      for method, details in methods : {
        path       = path
        method     = upper(title(method))
        integration = details["x-amazon-apigateway-integration"]
        name = regex("function:([^/]+)", details["x-amazon-apigateway-integration"].uri)[0]
      }
    ]
  ])
}
```

**Comment by bendrucker** [comment, marked as answer]
No, there is no way to bypass evaluating an expression. 

If you're getting an error, it's because you're passing bad input to `regex`. It has nothing to do with that specific rule, beyond the fact that it references this particular invalid expression. If it were _unknown_ input the function would not get evaluated. 

> The terraform module itself works totally fine as well, ie validate, plan, and apply all work fine.

If it were called the same way, this cannot be true. So you're somehow calling the module differently via TFLint, e.g., setting different variables, expecting data sources and then falling back on invalid defaults, etc.

**Reply by kferrone** [reply]
Oh would you look at that, it is looking at the file. I was setting an invalid default inside the file just like you say. Everything is working when the module is implemented with real values so in prod nothing ever uses that default file, so I missed that. 

Well thank you so much.
